### PR TITLE
fix: init.image.pullPolicy and nodeAgent.image.pullPolicy not being respect

### PR DESCRIPTION
**Pull request by zyue110026**
**What type of PR is this?**  bug

**Which issue does this PR fix?**:

#3009 

**What does this PR do / Why do we need it?**:
respect init.image.pullPolicy and nodeAgent.image.pullPolicy

**Testing done on this change**:

**Will this PR introduce any new dependencies?**:

**Will this break upgrades or downgrades? Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by yash97 — APPROVED** [review]
LGTM. I will wait for another approval.

**Review by orsenthil — APPROVED** [review]
Thank you for the pull request, and the bug report.

**Pull request merged**
